core/arch/arm/kernel/delay.c

```diff
@@ -34,7 +34,7 @@ void udelay(uint32_t us)
 	uint64_t start, target;
 
 	start = read_cntpct();
-	target = read_cntfrq() / 1000000ULL * us;
+	target = ((uint64_t)read_cntfrq() * us) / 1000000ULL;
```

Comment:
Should `MUL_OVERFLOW()` be used here perhaps?

Reply:
`us` is 32bit as `read_cntfrq()` and we're computing a 64bit value to be compared with cntpct.
No overflow expected.